src/decoder/lattice-incremental-decoder.h

```diff
@@ -0,0 +1,652 @@
+  BaseFloat beam_delta; // has nothing to do with beam_ratio
+  BaseFloat hash_ratio;
+  BaseFloat
+      prune_scale; // Note: we don't make this configurable on the command line,
+                   // it's not a very important parameter.  It affects the
+                   // algorithm that prunes the tokens as we go.
+  // Most of the options inside det_opts are not actually queried by the
+  // LatticeIncrementalDecoder class itself, but by the code that calls it, for
+  // example in the function DecodeUtteranceLatticeIncremental.
+  int32 max_word_id; // for GetLattice
+  fst::DeterminizeLatticePhonePrunedOptions det_opts;
+
+  LatticeIncrementalDecoderConfig()
+      : beam(16.0),
+        max_active(std::numeric_limits<int32>::max()),
+        min_active(200),
+        lattice_beam(10.0),
+        prune_interval(25),
+        determinize_delay(25),
+        determinize_max_active(std::numeric_limits<int32>::max()),
+        redeterminize(false),
+        redeterminize_max_frames(std::numeric_limits<int32>::max()),
+        epsilon_removal(false),
+        beam_delta(0.5),
+        hash_ratio(2.0),
+        prune_scale(0.1),
+        max_word_id(1e7) {}
+  void Register(OptionsItf *opts) {
+    det_opts.Register(opts);
+    opts->Register("beam", &beam, "Decoding beam.  Larger->slower, more accurate.");
+    opts->Register("max-active", &max_active,
+                   "Decoder max active states.  Larger->slower; "
+                   "more accurate");
+    opts->Register("min-active", &min_active, "Decoder minimum #active states.");
+    opts->Register("lattice-beam", &lattice_beam,
+                   "Lattice generation beam.  Larger->slower, "
+                   "and deeper lattices");
+    opts->Register("prune-interval", &prune_interval,
+                   "Interval (in frames) at "
+                   "which to prune tokens");
+    opts->Register("determinize-delay", &determinize_delay,
+                   "delay (in frames, typically larger than --prune-interval) "
+                   "at which to incrementally determinize lattices.");
+    opts->Register("determinize-max-active", &determinize_max_active,
+                   "This option is to adaptively decide the size of the chunk "
+                   "to be determinized. "
+                   "If the number of active tokens(in a certain frame) is less "
+                   "than this number (typically 50), we will start to "
+                   "incrementally determinize lattices from the last frame we "
+                   "determinized up to this frame. It can work with "
+                   "--determinize-delay to further reduce the computation "
+                   "introduced by incremental determinization. ");
+    opts->Register("redeterminize", &redeterminize,
+                   "whether to re-determinize the lattice after incremental "
+                   "determinization.");
+    opts->Register("redeterminize_max_frames", &redeterminize_max_frames,
```

Comment:
I'd add `Smaller -> less deterministic but less likely to blow up in bad cases.`